camel/societies/workforce/task_channel.py

```
@@ -180,3 +204,18 @@ async def get_channel_debug_info(self) -> str:
+        
+        Args:
+            timeout (float): max wait time (seconds)
+            context (str): context information, used for error messages
+            
+        Raises:
+            asyncio.TimeoutError: if wait times out
+        """
+        try:
+            await asyncio.wait_for(self._condition.wait(), timeout=timeout)
+        except asyncio.TimeoutError:
+            raise asyncio.TimeoutError(f"Timed out while {context}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)